File: OOP/Lab_1/lab_1.cpp
#include <iostream>
#include <array>
using namespace std;
class lab_1 {
private:
  int first;
  int second;
public:
  lab_1(){                                          //Constructor
      first= 0;
      second = 0;
  }
  void setValue(){
      cout << "first: ";
      cin >> first;
      cout << "second: ";
      cin >> second;
  }
  void getValue(){
      if (second<-1) cout << first<< "+" << -second << "i" << endl;
      else if (second==-1) cout << first << "+i" << endl;
      else if (second==0) cout << first << endl;
      else if (second==1) cout << first << "-i" << endl;
      else cout << first << "-" << second << "i" << endl;
  }
  void sqr(){
    first=first*first-(second*second);
    second=first*second*2;
  }
};
int main(){
  lab_1 obj1;
  lab_1 obj2;
  obj1.setValue();
  obj1.getValue();
  obj2.setValue();
  cout<<"Before:"<<endl;
  obj2.getValue();
  obj2.sqr();
  cout<<"After:"<<endl;
  obj2.getValue();
  array <lab_1, 2> objArray={obj1, obj2};
  objArray[1].getValue();
  objArray[1].setValue();
  objArray[1].getValue();
  return 0;
}

File: OOP/Lab_1/lab_1.java
import java.util.Scanner;
public class lab_1 {
  int first;
  int second;
  public static Scanner input = new Scanner(System.in);
  public lab_1(int first, int second){//Constructor
    this.first=first;
    this.second=second;
  }
  public void setValue(){
        System.out.println("first: ");
        first=input.nextInt();
        System.out.println("second: ");
        second=input.nextInt();
    }
    public void getValue(){
        if (second<-1) System.out.println(first + "+" + -second + "i");
        else if (second==-1) System.out.println(first + "+i");
        else if (second==0) System.out.println(first);
        else if (second==1) System.out.println(first + "-i");
        else System.out.println(first + "-" + second + "i");
    }
    public void sqr(){
      first=first*first-(second*second);
      second=first*second*2;
    }
  public static void main(String[]args){
    lab_1 obj1 = new lab_1(0,0);
    lab_1 obj2 = new lab_1(0,0);
    obj1.setValue();
    obj1.getValue();
    obj2.setValue();
    System.out.println("Before:");
    obj2.getValue();
    obj2.sqr();
    System.out.println("After:");
    obj2.getValue();
    //array
  }
}
